pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -127,9 +127,25 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 		return api.Environment{}, err
 	}
 	setProductLabels(cr, &mergedEnv)
+	setCRBuildConfigEnvsToSpec(cr, &mergedEnv)
 	return mergedEnv, nil
 }
 
+func setCRBuildConfigEnvsToSpec(cr *api.KieApp, env *api.Environment) {
+	if len(env.Servers) > 0 {
+		for index := range cr.Spec.Objects.Servers {
+			for bcindex, srv := range env.Servers[index].BuildConfigs {
+				if srv.Spec.Strategy.SourceStrategy.Env != nil && cr.Spec.Objects.Servers[index].Build != nil {
+					for _, crenv := range cr.Spec.Objects.Servers[index].Build.Envs {
+						env.Servers[index].BuildConfigs[bcindex].Spec.Strategy.SourceStrategy.Env = append(
+							env.Servers[index].BuildConfigs[bcindex].Spec.Strategy.SourceStrategy.Env, crenv)
+					}
```

Comment:
we already have a method that i believe will do what you need ... 
https://github.com/kiegroup/kie-cloud-operator/blob/d3cab494497a791b417306581623963cea7b2604/pkg/controller/kieapp/defaults/defaults.go#L636
https://github.com/kiegroup/kie-cloud-operator/blob/d3cab494497a791b417306581623963cea7b2604/pkg/controller/kieapp/shared/utils.go#L145-L156

could it maybe go here?
https://github.com/kiegroup/kie-cloud-operator/blob/d3cab494497a791b417306581623963cea7b2604/pkg/controller/kieapp/defaults/defaults.go#L625-L628

Reply:
The consolidateObjects will be called only during the reconcile, wouldn't better configure it before?

Reply:
update the func to use EnvOverride.

Reply:
everything the operator does is ultimately only called during the reconcile. my main concern is to keep alike code together so we can find things more easily

Reply:
I see, updated.